File: android/src/main/java/com/reactnativehaptics/HapticsType.kt
package com.reactnativehaptics

/**
 * This defines the different types
 * of haptics that we support
 * throughout the package.
 *
 * @author Jason Kelly <jason-kelly@live.dk>
 */
enum class HapticsType(private val value: String) {
  SELECTION("selection"),
  IMPACT_LIGHT("impactLight"),
  IMPACT_MEDIUM("impactMedium"),
  IMPACT_HEAVY("impactHeavy"),
  NOTIFICATION_SUCCESS("notificationSuccess"),
  NOTIFICATION_WARNING("notificationWarning"),
  NOTIFICATION_ERROR("notificationError");

  override fun toString(): String {
    return this.value
  }
}

File: ios/HapticsType.swift
//
//  HapticsType.swift
//  Haptics
//
//  Created by Anders Friis on 16/02/2020.
//  Copyright © 2020 Facebook. All rights reserved.
//

import Foundation
import UIKit

enum HapticsType: String, CaseIterable {
    case selection, impactLight, impactMedium, impactHeavy, notificationSuccess, notificationWarning, notificationError
    
    var feedbackType: FeedbackType {
        switch self {
        case .selection: return .selection
        case .impactLight, .impactMedium, .impactHeavy:
            var type = UIImpactFeedbackGenerator.FeedbackStyle.light
            
            switch self {
            case .impactLight:
                type = .light
            case .impactMedium:
                type = .medium
            case .impactHeavy:
                type = .heavy
            default: break
            }
            return .impact(style: type)
            
        case .notificationSuccess, .notificationWarning, .notificationError:
            var type = UINotificationFeedbackGenerator.FeedbackType.success
            
            switch self {
            case .notificationSuccess:
                type = .success
            case .notificationWarning:
                type = .warning
            case .notificationError:
                type = .error
            default: break
            }
            return .notification(type: type)
        }
    }
    
}

enum FeedbackType {
    case selection, impact(style: UIImpactFeedbackGenerator.FeedbackStyle), notification(type: UINotificationFeedbackGenerator.FeedbackType)
}
